### ISSUE TITLE
run doesn't run things properly

### ISSUE THREAD

**Issue by withinboredom** [yarnpkg/yarn#911]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
``` sh
yarn run build
$ "./go.sh build"
sh: ./go.sh build: No such file or directory
error Command failed with exit code 127.
info Visit http://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**If the current behavior is a bug, please provide the steps to reproduce.**
Create a `package.json` with the following:

``` json
{
  "scripts": {
    "build" : "./go.sh build"
}
```

and a file called `go.sh` in the same directory:

``` sh
echo "hello world"
echo "You typed: $@"
```

**What is the expected behavior?**
Output to be similar to:

```
hello world
you typed: build
```

**Please mention your node.js, yarn and operating system version.**
yarn: 0.15.1

This works fine in 0.15.0

**Comment by icd2k3** [comment]
This should be fixed in https://github.com/yarnpkg/yarn/pull/809

**Comment by kittens** [comment]
Yup, fixed in #809.
